list-agreements.lua

```diff
@@ -41,19 +40,50 @@ function get_petition_date(agr)
     julian_day = math.floor(agr_year_tick / 1200) + 1
     agr_month = math.floor(julian_day / 28) + 1
     agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
+end
+
+function get_petition_age(agr)
+    agr_year_tick = agr.details[0].year_tick
+    agr_year = agr.details[0].year
+    cur_year_tick = df.global.cur_year_tick
+    cur_year = df.global.cur_year
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
+        del_year_tick = agr_year_tick - cur_year_tick
+    end
+    julian_day = math.floor(del_year_tick / 1200) + 1
+    del_month = math.floor(julian_day / 28)
+    del_day = julian_day % 28
+    return {del_year,del_month,del_day}
 end
 
 function get_guildhall_profession(agr)
     prof = agr.details[0].data.Location.profession
     profname = string.lower(df.profession[prof])
+    -- *VERY* important code follows
+    if string.find(profname, "man") then
+        profname = string.gsub(profname,"man",string.lower(dfhack.units.getRaceNameById(df.global.ui.race_id)))
+    end
     return profname:gsub("_", " ")
 end
 
-function get_religion_name(agr)
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
+    party_id = agr.parties[0].entity_ids[0]
+    party_name = dfhack.TranslateName(df.global.world.entities.all[party_id].name, true)
+    return party_name
+end
+
+function get_deity_name(agr)
     religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+    for _,deity_id in ipairs(df.global.world.entities.all[religion_id].relations.deities) do
+        return dfhack.TranslateName(df.global.world.history.figures[deity_id].name,true)
+    end
```

Comment:
is this written this way because `deities` can be an empty list? The calling code doesn't appear to handle `nil` as a return value, so perhaps this should be written as:
```
local deities = df.global.world.entities.all[religion_id].relations.deities
if #deities == 0 then return '' end
return dfhack.TranslateName(df.global.world.history.figures[deities[0]].name,true)
```

Reply:
This was because i was getting `nil` errors when trying to read the value from `.deities[0]` but handled fine when parsing the array. `.deities` is, to my knowledge, always a single element array for religions, so reading `.deities[0]` *should* be viable.

This was the workaround that generated the desired outcome

Reply:
That seems.. strange. I'll have to look into the structures involved here.
btw, if the array *is* empty, then the calling code will error out when this function returns nil. maybe add a `return 'Unknown deity'` after the loop?